extension-manifest-v3/src/background/reporting.js

```diff
@@ -267,6 +273,20 @@ function onLocationChange(details) {
     }
 
     try {
+      if (pendingStartup) {
+        await Promise.race([
```

Comment:
This might be a good pattern to add in `adblocker.js` as if indeed promise hangs, the events will be queuing...